openra/containers.py

```diff
@@ -29,3 +39,26 @@ class Container(containers.DeclarativeContainer):
             settings.GITHUB_API_KEY
         )
     )
+
+    engine_file_repository = providers.Singleton(
+        EngineFileRepository
+    )
+
+    map_file_repository = providers.Singleton(
+        MapFileRepository
+    )
+
+    file_downloader = providers.Singleton(
+        FileDownloader
+    )
+
+    utility = providers.Singleton(
+        Utility
+    )
+
+    map_search = providers.Singleton(
+        MapSearch
```

Comment:
There are tests that don't use this and other registered services via DI, but instance them manually!

Reply:
That should only be in the tests of the class itself. For example in the test for MapSearch, it will create a MapSearch instance directly so that it can control the parameters passed in, and will likely test it with fakes or mocks.

Then it can use the fakes or mocks to make sure that the right functions were called with the correct parameters.

In the route that uses these, they would probably be swapped out for mocks or fakes, and just check that the view called the right functions with the right parameters, and maybe trigger each of the different response types to make sure the route works correctly.

There isn't currently a test that tests that the DI is working, other than using it to swap in mocks or fakes.

I could add a set of tests for that, that goes through and makes sure each of these objects get created correctly?

Reply:
Ok I've added some tests for the DI container now to make sure that that stays all working correctly.